backend/Code/FrequencyScores.py

```diff
@@ -0,0 +1,128 @@
+from bs4 import BeautifulSoup
+import requests
+from dicttoxml import dicttoxml
+from xml.dom.minidom import parseString
+import xmltodict
+import os
+
+# Class that retrieves the score of a letter in a specified language according
+# to its frequency
+
+# Base url of the requests
+BASE_URL = "http://www.sttmedia.com/"
+CHARS = "characterfrequency"
+SYLLABLES = "syllablefrequency"
+
+# Select the min and max score values possible for any letter
+DEF_MIN = 1
+DEF_MAX = 20
+
+# Language from which to retrieve the letter scores (as it appears
+# in the web page)
+LANG = "nederlands"
+FOLDER = LANG + "Scores"
+
+# This is the main function for making queries.
+# An html document should be returned by the query.
+def query_site(url):
+    r = requests.get(url)
+    print ("requesting", r.url)
+
+    if r.status_code == requests.codes.ok:
+        return BeautifulSoup(r.text, "html.parser")
+    else:
+        r.raise_for_status()
+
+# This adds a language to the url before making
+# an API call to the function above.
+def query_by_lang(url, lang):
+    url = url + '-' + lang
+    return query_site(url)
+
+# Functions to get the max and min value of a list previously sorted
+# from greater to smaller
+def find_min(list):
+    # Get last element of the list
+    return find_value_at(list, -1)
+
+def find_max(list):
+    # Get first element of the list
+    return find_value_at(list, 0)
+
+# Returns the value at position indx of list
+def find_value_at(list, indx):
+    row = list[indx]
+    val = row.find_all("td")[1].string[:-1]
+    return float(val.replace(',', '.'))
+
+# Maps val which is in a range of o_min to o_max to a new range between
+# n_min and n_max, making o_min map to n_max and o_max to n_min
+# (The higher val the smaller the new value)
+def map_to_range(val, o_min, o_max, n_min, n_max):
+    new_value = n_max - (val - o_min) * (n_max - n_min) / (o_max - o_min)
+    return new_value
+
+# Returns a dictionary with the letters as keys and their scores as values
+def create_dict(list, min, max, MIN, MAX):
```

Comment:
1) Having two min and two max is confusing. 2) Capitals are used for global constants which they are not